db/migrate/20200508234114_add_indexed_to_rubygems.rb

```diff
@@ -0,0 +1,10 @@
+class AddIndexedToRubygems < ActiveRecord::Migration[6.0]
+  def change
+    add_column :rubygems, :indexed, :boolean, null: false, default: false
+    add_index :rubygems, :indexed
+
+    say_with_time "populating indexed rubygems table column" do
+      Rubygem.joins(:versions).where(versions: {indexed: true}).update_all(indexed: true)
```

Comment:
Will it be updating the same rubygem multiple times? Would adding distinct help here?
```
Rubygem.joins(:versions).where(versions: {indexed: true}).count
   (373.4ms)  SELECT COUNT(*) FROM "rubygems" INNER JOIN "versions" ON "versions"."rubygem_id" = "rubygems"."id" WHERE "versions"."indexed" = $1  [["indexed", true]]
 => 1062022
```

Reply:
IMHO no, PostgreSQL is clever enough to do the most effective way on its own. Using DISTINCT actually increase the query "cost" here.